src/commands/legacy/strike.js

```diff
@@ -147,7 +147,18 @@ command.executePunishment = async (punishment, guild, user, bot, database, reaso
   let member;
   switch (punishment.action) {
     case 'ban':
-      await ban.ban(guild, user, bot.user, reason, punishment.duration);
+      const ban = new Ban(/** @type {module:"discord.js".Message}*/{
```

Comment:
You should probably separate the Ban itself from the command. "Faking" a command call is not really a clean solution.